src/useradd.c

```diff
@@ -542,7 +542,7 @@ static int set_defaults (void)
 	char  *default_file = USER_DEFAULTS_FILE;
 	char  *cp;
 	FILE  *ifp;
-	FILE  *ofp;
+	FILE  *ofp = NULL;
```

Comment:
`NULL` is not valid input to *fclose*(3).

```c
$ grepc fclose /usr/include/
/usr/include/stdio.h:extern int fclose (FILE *__stream) __nonnull ((1));
```

Reply:
I wasn't aware of this. Thanks for pointing me in the right direction.